tests/bgp/test_bgp_bbr.py

```diff
@@ -0,0 +1,273 @@
+    tor1_offset = tbinfo['topo']['properties']['topology']['VMs'][tor1]['vm_offset']
+    tor1_exabgp_port = EXABGP_BASE_PORT + tor1_offset
+    tor1_exabgp_port_v6 = EXABGP_BASE_PORT_V6 + tor1_offset
+
+    dut_asn = tbinfo['topo']['properties']['configuration_properties']['common']['dut_asn']
+    tor1_asn = nbrhosts[tor1]['conf']['bgp']['asn']
+    aspath = '{} {}'.format(dut_asn, DUMMY_ASN1)
+    aspath_dual_dut_asn = '{} {} {} {}'.format(dut_asn, DUMMY_ASN1, dut_asn, DUMMY_ASN2)
+
+    Route = namedtuple('Route', ['prefix', 'nexthop', 'aspath'])
+
+    bbr_route = Route(BBR_PREFIX, neigh_peer_map[tor1]['peer_addr'], aspath)
+    bbr_route_v6 = Route(BBR_PREFIX_V6, neigh_peer_map[tor1]['peer_addr_v6'], aspath)
+
+    bbr_route_dual_dut_asn = Route(BBR_PREFIX, neigh_peer_map[tor1]['peer_addr'], aspath_dual_dut_asn)
+    bbr_route_v6_dual_dut_asn = Route(BBR_PREFIX_V6, neigh_peer_map[tor1]['peer_addr_v6'], aspath_dual_dut_asn)
+
+    setup_info = {
+        'tor1': tor1,
+        'other_vms': other_vms,
+        'tor1_offset': tor1_offset,
+        'tor1_exabgp_port': tor1_exabgp_port,
+        'tor1_exabgp_port_v6': tor1_exabgp_port_v6,
+        'dut_asn': dut_asn,
+        'tor1_asn': tor1_asn,
+        'bbr_route': bbr_route,
+        'bbr_route_v6': bbr_route_v6,
+        'bbr_route_dual_dut_asn': bbr_route_dual_dut_asn,
+        'bbr_route_v6_dual_dut_asn': bbr_route_v6_dual_dut_asn
+    }
+
+    logger.info('setup_info: {}'.format(json.dumps(setup_info, indent=2)))
+
+    return setup_info
+
+
+def update_routes(action, ptfip, port, route):
+    if action == 'announce':
+        msg = '{} route {} next-hop {} as-path [ {} ]'.format(action, route.prefix, route.nexthop, route.aspath)
+    elif action == 'withdraw':
+        msg = '{} route {} next-hop {}'.format(action, route.prefix, route.nexthop)
+    else:
+        logger.error('Unsupported route update operation.')
+        return
+    url = 'http://%s:%d' % (ptfip, port)
+    data = {'commands': msg }
+    r = requests.post(url, data=data)
+    assert r.status_code == 200
+
+
+@pytest.fixture
+def announce_withdraw_route(setup, ptfhost):
+    tor1_exabgp_port = setup['tor1_exabgp_port']
+    tor1_exabgp_port_v6 = setup['tor1_exabgp_port_v6']
+    bbr_route = setup['bbr_route']
+    bbr_route_v6 = setup['bbr_route_v6']
+
+    logger.info('Announce route {} and {} to the first T0'.format(str(bbr_route), str(bbr_route_v6)))
+    update_routes('announce', ptfhost.mgmt_ip, tor1_exabgp_port, bbr_route)
+    update_routes('announce', ptfhost.mgmt_ip, tor1_exabgp_port_v6, bbr_route_v6)
+    time.sleep(3)
+
+    yield bbr_route, bbr_route_v6
+
+    logger.info('Withdraw route {} and {} from the first T0'.format(str(bbr_route), str(bbr_route_v6)))
+    update_routes('withdraw', ptfhost.mgmt_ip, tor1_exabgp_port, bbr_route)
+    update_routes('withdraw', ptfhost.mgmt_ip, tor1_exabgp_port_v6, bbr_route_v6)
+
+
+@pytest.fixture
+def announce_withdraw_dual_dut_asn_route(setup, ptfhost):
```

Comment:
Does it repeat the previous function?
Should we reuse the previous one?

Reply:
Sure, updated, thanks!